src/Val.h

```diff
@@ -1245,37 +1299,77 @@ class ZValElement {
      * Initialize a ZValElement using a TypeDecl assignment.
      *
      * This is used at record construction time to set the is_managed
-     * and tag fields properly.
+     * and tag fields properly and get the ZValElement into the unset
```

Comment:
Micro-nit: structures like "foo and bar and bletch" read better as "foo, bar and bletch". (I will also be okay with an Oxford comma there ;-).